app/services/google_analytics_service.rb

```diff
@@ -3,7 +3,19 @@ def client
     @client ||= GoogleAnalytics::Client.new.build
   end
 
-  def page_views(base_paths)
+  def pageviews_for_date(date, *base_paths)
+    formatted_date = date.strftime("%Y-%m-%d")
+
+    request = GoogleAnalytics::Requests::PageViewsRequest.new.build(
+      base_paths: base_paths,
+      start_dates: [formatted_date],
+      end_date: formatted_date
+    )
+    response = client.batch_get_reports(request)
```

Comment:
We are using a `batch_request` but we are not getting the most of it as we are only passing 1 Report Request. We are only retrieving data for 1 day, so we should be able to include until 5 reports with hundreds/thousands of paths in each request.

Reply:
We are also impliciting batching to 500 paths per request, which is not much considering that we only request data for 1 day.